### PR TITLE
[CELEBORN-316] Wrap Celeborn exception with CelebornIOException

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
When we enable spark blacklist mechanism, many executor was excluded from current stage caused by RSS exception, but for all RSS exception we should not been excluded and this mechanism caused less and less executor to return the task, caused job slower. 

We should wrap RSS exception with CelebonrIOException then we can do something in spark side

### CODE REVIEW

**Comment by RexXiong** [comment]
Can we also change the related methods signature from IOException -> CelebornIOException

**Comment by AngersZhuuuu** [comment]
> Can we also change the related methods signature from IOException -> CelebornIOException

I think it's unnecessary, some place extended from spark's API

File: common/src/main/scala/org/apache/celeborn/common/exception/CelebornIOException.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.exception
+
+import java.io.IOException
+
+class CelebornIOException(message: String, cause: Throwable)
```
Comment (FMX) [inline common/src/main/scala/org/apache/celeborn/common/exception/CelebornIOException.scala:22]: How about CelebornException? There might be something like register failed or sendRpc Exception. If you wanna handle celeborn exceptions in spark, I think there is no reason to assume that all exceptions are IO exceptions.

Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/exception/CelebornIOException.scala:22]: > How about CelebornException? There might be something like register failed or sendRpc Exception. If you wanna handle celeborn exceptions in spark, I think there is no reason to assume that all exceptions are IO exceptions.

It should extended from IOException

**Review by FMX — APPROVED** [review]
LGTM.Thanks.

**Pull request merged**